source/Commands/CommandObjectExpression.cpp

```diff
@@ -398,9 +398,50 @@ bool CommandObjectExpression::EvaluateExpression(const char *expr,
       options.SetTimeout(std::chrono::microseconds(m_command_options.timeout));
     else
       options.SetTimeout(llvm::None);
-
-    ExpressionResults success = target->EvaluateExpression(
-        expr, frame, result_valobj_sp, options, &m_fixed_expression);
+    
+    // Swift does not have an IR interpreter, so expression evaluation for
+    // simple variable access is quite slow.  Accelerate that by first seeing
+    // if the whole expression is a simple identifier, and if so look it up
+    // as a local variable.
+    // Note, Swift's type resolution rules are complex enough that even figuring
+    // out whether the dynamic and static versions of foo.bar are the same is
+    // tricky, so we only handle single identifiers.
+    // Also, we don't do this if we are using the REPL, top level code, 
+    // or if the PlaygroundTransform is enabled.
+    ExpressionResults success = eExpressionSetupError;
+    if (target->GetUseFrameVarToAccelerateExpr()
+        && !target->UseScratchTypesystemPerModule()
+        && frame
+        && (options.GetLanguage() == eLanguageTypeSwift
+                  || (options.GetLanguage() == eLanguageTypeUnknown
+                      && frame->GetLanguage() == eLanguageTypeSwift))
+              && options.GetExecutionPolicy() != eExecutionPolicyTopLevel
+              && !options.GetREPLEnabled()
+              && !options.GetPlaygroundTransformEnabled())
+    {
+        if (swift::Lexer::isIdentifier(expr))
+        {
+            VariableSP var_sp;
+            Status error;
+            result_valobj_sp
+                = frame->GetValueForVariableExpressionPath(expr,
+                                                          eDynamicCanRunTarget,
+                                                          0,
+                                                          var_sp,
+                                                          error);
+            if(error.Success()
+               && result_valobj_sp
+               && result_valobj_sp->GetError().Success())
+            {
+              success = eExpressionCompleted;
+              result_valobj_sp = result_valobj_sp->Persist();
```

Comment:
Would it make sense to early-exit on success, perhaps by factoring out this and the regular evaluation into a separate function? I always find the control flow in these multi-page functions incredibly difficult to follow otherwise.

Reply:
You can't early exit on success, since you need to do all the work from line 455 following in both cases.  The regular evaluation is two lines of code, I don't think it would be clearer to make that a separate function.  But I'll make the swift local lookup a separate function.

Reply:
I guess what we meant was:

```
Status fastPath(...) {
  if (cond1 || cond2 || cond3)
    return NotExecuted
  Evaluate
  return evaluate status
}
```

I really find nested control flow easy to get wrong and super-hard to read.

Reply:
Yes, I'll do that when I extract this code into a separate function.